Minio.Benchmarks/BenchmarkGetRegionFromEndpoint.cs

```diff
@@ -0,0 +1,59 @@
+﻿/*
+* MinIO .NET Library for Amazon S3 Compatible Cloud Storage,
+* (C) 2017, 2018, 2019, 2020 MinIO, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using BenchmarkDotNet.Attributes;
+
+namespace Minio.Benchmarks
```

Comment:
Looks like these benchmark runs will not use cache?

Reply:
And in codebase, we use only pre-compiled regex. Why do we need others here?

Reply:
The method being benchmarked was copied from how the code was.

Reply:
@ngbrown , can we leave the Benchmark csproj out of the PR? Other than that, LGTM

Reply:
I rebased the whole pull request and removed the benchmark project.